File: 0011-container-with-most-water/0011-container-with-most-water.ts
function maxArea(height: number[]): number {
    let max_area = 0, a = 0, b = height.length - 1;
    while(a < b){
        let temp_area = Math.min(height[a], height[b])*(b-a);
        max_area = Math.max(temp_area, max_area);
        if(height[a] <= height[b]){
            a++;
        } else {
            b--;
        }
    }
    return max_area;
};

File: 0011-container-with-most-water/0011-container-with-most-water.rs
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let (mut max, mut a, mut b) = (0, 0, height.len() - 1);
        
        while a < b {
            let min_height = std::cmp::min(height[a], height[b]);
            max = std::cmp::max(max, min_height * (b - a) as i32);
            
            if height[a] <= height[b] {
                a += 1;
            } else {
                b -= 1;
            }
        }
        
        max
    }
}